server/engine/src/main/java/org/eclipse/lsp/cobol/service/copybooks/CopybookServiceImpl.java

```diff
@@ -288,10 +288,10 @@ public void sendCopybookDownloadRequest(String documentUri, Collection<String> c
       LOG.debug("Copybooks to download: {}", copybooksToDownload);
       if (!copybooksToDownload.isEmpty()) {
         clientProvider.get().downloadCopybooks(documentUri,
-            copybooksToDownload.stream().map(CopybookName::getQualifiedName).collect(toList()),
-            Optional.ofNullable(copybooksToDownload.stream().findFirst().get().getDialectType()).orElse(COBOL), //NOSONAR
+                copybooksToDownload.stream().map(CopyBookDTO::new).collect(toList()),
```

Comment:
You might be able to even move the map after the `flatMap(Set::stream)` above and avoid constructing the intermediate list.